doc/rpc.rst

```diff
@@ -0,0 +1,54 @@
+.. _rpc:
+
+********
+Dune RPC
+********
+
+Starting from dune 3.0, dune's watch mode also runs an RPC server. This is a
+general mechanism introduced to integrate with various dune functionality. Some
+use cases we have in mind are:
+
+- Text editors that would like to receive dune's error reporting
+- Custom actions that need to communicate with dune
```

Comment:
This second point should not be tied to the watch mode, otherwise it means that you couldn't use such actions in normal builds. Perhaps:

```suggestion
- In the future, custom actions that need to communicate with dune
```

To make it clear we are not there yet and so the design of this part should be taken with a grain of salt